### PR TITLE
[2.0.x] Improved Pause/Load/Unload, M701-M702

### PR DESCRIPTION
**Pull request by thinkyhead**
Based on #7107 by @tcm0116 

- Adds `M701` / `M702` general codes to load and unload filament.
- MK2 multiplexer (SNMM) is supported via `tool_change`.

### CODE REVIEW

**Comment by petrzjunior** [comment]
Wow, good job @tcm0116 and @thinkyhead! Can't wait to test it.

File: Marlin/src/feature/pause.h [hunk 1]
```diff
@@ -30,14 +30,49 @@
...
 
 void wait_for_filament_reload(const int8_t max_beep_count=0);
 
-void resume_print(const float &load_length=0, const float &initial_extrude_length=0, const int8_t max_beep_count=0);
+void resume_print(const float &load_length=0, const float &extrude_length=ADVANCED_PAUSE_EXTRUDE_LENGTH, const int8_t max_beep_count=0);
+
+static bool load_filament(const float &load_length=0, const float &extrude_length=0, const int8_t max_beep_count=0, const bool show_lcd=false,
+                          const bool pause_for_user=false, const AdvancedPauseMode mode=ADVANCED_PAUSE_MODE_PAUSE_PRINT);
+
+static bool unload_filament(const float &unload_length, const bool show_lcd=false, const AdvancedPauseMode mode=ADVANCED_PAUSE_MODE_PAUSE_PRINT);
```
Comment (tcm0116) [inline Marlin/src/feature/pause.h]: `load_filament` and `unload_filament` shouldn't be declared as `static`.

File: Marlin/src/feature/pause.cpp [hunk 2]
```diff
@@ -70,91 +75,216 @@ static float resume_position[XYZE];
...
   set_destination_from_current();
-  #if IS_KINEMATIC
-    planner.buffer_line_kinematic(destination, fr, active_extruder);
+  destination[E_AXIS] += length / planner.e_factor[active_extruder];
+  buffer_line_to_destination(fr);
+  stepper.synchronize();
+  set_current_from_destination();
+}
+
+static bool load_filament(const float &load_length/*=0*/, const float &extrude_length/*=0*/, const int8_t max_beep_count/*=0*/,
```
Comment (tcm0116) [inline Marlin/src/feature/pause.cpp]: `load_filament` shouldn't be declared as `static`.

File: Marlin/src/feature/pause.cpp [hunk 3]
```diff
@@ -70,91 +75,216 @@ static float resume_position[XYZE];
...
+      show_lcd && advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE
+    #else
+      0
+    #endif
+  );
+
+  return true;
+}
+
+static bool unload_filament(const float &unload_length, const bool show_lcd/*=false*/,
```
Comment (tcm0116) [inline Marlin/src/feature/pause.cpp]: `unload_filament` shouldn't be declared as `static`.

**Review by tcm0116 — CHANGES_REQUESTED** [review]

**Comment by thinkyhead** [comment]
@petrzjunior — Have you been using the `bugfix-2.0.x` branch? This stuff is already merged into `bugfix-1.1.x` and should be ready to go.

**Comment by tcm0116** [comment]
Awesome! Thanks for porting this over!

**Pull request merged**